src/sql/src/func.rs

```diff
@@ -287,6 +287,24 @@ pub struct FuncImpl<R> {
     op: Operation<R>,
 }
 
+pub trait FuncImplCatalogDetails {
```

Comment:
Does this trait need to exist? Seems like you could just export these as inherent methods and use directly from `coord`.

Reply:
The problem is that each `sql::func::Func` has a different `R` in its `FuncImpl<R>`, so I don't know what type I could return that would let `coord` access these functions. If you have an idea, I'm very game because this feels very oblique.

Reply:
I do have an idea, actually! Can you just implement these things on `enum Func`, and match your way down into the `oid`/`params` fields?

Reply:
The problem is that we want the details from each internal `FuncImpl`s, not the `Func` itself. We could match on the internal `FuncImpl`, but we still need to aggregate them into something that we return. Rather than doing the admittedly byzantine trait, I instead just put these things in a `struct`. lmk if that's cleaner of if there's a cleaner solution you're seeing that I'm not.